planner/core/task.go

```diff
@@ -1256,6 +1257,13 @@ func (p *PhysicalLimit) sinkIntoIndexLookUp(t task) bool {
 	if !isTableScan {
 		return false
 	}
+	// We can't sink Limit into IndexLookUpReader when it scans a partition table in dynamic prune mode
+	// https://github.com/pingcap/tidb/issues/32516
+	if table, ok := p.ctx.GetInfoSchema().(infoschema.InfoSchema).TableByID(ts.Table.ID); ok {
```

Comment:
It seems that `GetInfoSchema` is deprecated. Can we use  `TxnManager.GetTxnInfoSchema` instead?